### PR TITLE
Don't show the indicator on desktops that are only charging peripherals

### PR DESCRIPTION
**Pull request by davidmhewitt**
For example, at my desktop PC that doesn't have a battery, using my Wacom tablet wired, I still got the power indicator showing a fully charged battery (referring to my tablet).

It's a little confusing as it looks like the indicator is still trying to show the charge status of the computer as that's where I expect to see my global state icons.

To workaround this a little, this makes it so the indicator only appears if there is a battery that powers the computer (UPS or Laptop Battery) or if there are discharging peripherals.

If there are only charging peripherals and no battery that's capable of powering the computer, just don't display the indicator.

### CODE REVIEW

**Comment by danrabbit** [comment]
Hm I wonder if maybe we should add a symbolic version of `battery-ac-adapter` or something like this instead. I feel like being able to see that a device has been charged is probably important info still

**Comment by davidmhewitt** [comment]
Yeah, that's probably better. I'll leave this branch open for now since we can use largely the same logic to change the icon instead.

**Comment by danrabbit** [comment]
Proposed an icon here: https://github.com/elementary/icons/pull/619

File: src/Services/DeviceManager.vala [hunk 1]
```diff
@@ -117,6 +118,15 @@ public class Power.Services.DeviceManager : Object {
         });
 
         has_battery = batteries.has_next ();
+
+        // Only interested in devices if they're discharging peripherals or supplying power to the computer
+        only_charging_peripheral_devices = true;
+        foreach (var device in devices.values) {
+            if (device.power_supply || device.state == DEVICE_STATE_DISCHARGING || device.state == DEVICE_STATE_EMPTY) {
```
Comment (ryonakano) [inline src/Services/DeviceManager.vala]:
```suggestion
            if (device.power_supply || device.state == DISCHARGING || device.state == EMPTY) {
```

To reflect changes in [this commit](https://github.com/elementary/wingpanel-indicator-power/commit/aa920e13acbab3292b7e0422018d58ab1d380154#diff-e522e471998bdef45f66d6adffe919ad)

File: src/Indicator.vala [hunk 2]
```diff
@@ -110,8 +110,7 @@ public class Power.Indicator : Wingpanel.Indicator {
     private void show_display_device_data () {
         if (display_device != null && display_widget != null) {
             var icon_name = display_device.get_symbolic_icon_name_for_battery ();
-            display_widget.icon_name = icon_name;
-            display_widget.allow_percent = true;
+            display_widget.set_icon_name (icon_name, true);
```
Comment (ryonakano) [inline src/Indicator.vala]:
```suggestion
            display_widget.icon_name = icon_name;
            display_widget.allow_percent = true;
```

I don't think this change makes sense now.

**Comment by quequotion** [comment]
Is pull request #129 then superceding this?

**Comment by kvdz** [comment]
i agree with  @quequotion . i think we should close this pr (as in not merge)

**Pull request closed without merging**